src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -293,20 +294,19 @@ public boolean checkPathExists(@NonNull String branchOrHash, @NonNull String pat
     @CheckForNull
     @Override
     public String getDefaultBranch() throws IOException, InterruptedException {
-        String url = V1_API_BASE_URL + this.owner + "/" + this.repositoryName + "/main-branch";
+        String url = V2_API_BASE_URL + this.owner + "/" + this.repositoryName + "/?fields=mainbranch.name";
```

Comment:
does this work against Hg repos?

Reply:
Yes it does. I found a bug in my code thats fixed now.